starter-code/src/viking.js

```diff
@@ -1,11 +1,57 @@
-// Soldier
-function Soldier() {}
+// Soldier constructor
+function AbstractSoldier(healthArg, strengthArg) {
+    this.health = healthArg;
+    this.strength = strengthArg;
+}
+Soldier.prototype.attack = function () {
```

Comment:
Careful here, you changed the name of the constructor so every time you used Soldier, you needed to use AbstractSoldier